ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+
+        # Write to new setting file
+        lisvap_file = f"{self.work_dir}/lisvap_setting_{timestamp}.xml"
+        cfg.save(lisvap_file)
+        return lisvap_file
+
+    # TODO take this out of the class?
+    def run_lisvap(self, forcing: PathLike) -> Tuple[int, bytes, bytes]:
+        """Run lisvap to generate evaporation input files
+
+        Args:
+            forcing: Path to re-indexed forcings
+
+        Returns:
+            Tuple with exit code, stdout and stderr
+        """
+        singularity_image = CFG['lisflood.singularity_image']
+        docker_image = CFG['lisflood.docker_image']
+        self._check_forcing(forcing)
+        lisvap_file = self._create_lisvap_config()
+
+        mount_points = {
+            f'{self.parameterset.root}': '/data/lisflood_input',
+            f'{self.parameterset.mask}': '/data/model_mask.nc',
+            f'{self.forcing_dir}': '/data/forcing',
+            f'{self.work_dir}': '/output',
+        }
+
+        if CFG['container_engine'].lower() == 'singularity':
+            args = [
+                "singularity",
+                "exec",
+            ]
+            args += ["--bind", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(singularity_image)
+
+        elif CFG['container_engine'].lower() == 'docker':
+            args = [
+                "docker",
+                "run -ti",
+            ]
+            args += ["--volume", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(docker_image)
+
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+
+        lisvap_filename = Path(lisvap_file).name
+        args += ['python3', '/opt/Lisvap/src/lisvap1.py', f"/output/{lisvap_filename}"]
+        container = subprocess.Popen(args, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        exit_code = container.wait()
+        stdout, stderr = container.communicate()
+        return exit_code, stdout, stderr
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_y(grid),
+                "latitude": self.bmi.get_grid_x(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da
+
+    @property
+    def parameters(self) -> Iterable[Tuple[str, Any]]:
```

Comment:
I don't really understand why these would be the 'configurable' parameters. Wouldn't it make more sense if it listed the options in the lisflood config file?

Reply:
I agree, @sverhoeven what do you think?

Reply:
Now setup and parameters are sides of the same coin. So any change we make here should also be done in setup().
The forcing dir and working dir become part of the config file.

Which arguments should the setup have?  In wflow it is very dynamic but luckily 2 levels deep. In Lisflood there is a more complicated config with differently shaped sections and nestings.

Reply:
I'm not sure either. I just think the philosophy of PyMT is that these parameters are the things you'd like to change about the model configuration. So more like the default values from the [tables here](https://ec-jrc.github.io/lisflood-model/3_01_optLISFLOOD_overview/). But I agree that there is probably no neat way to do that right now. Perhaps one way would be to do it through the ParameterSet class you created. 

I think I'd prefer a NotImplementedError over the current solution, because I'm not very happy about e.g. having to pass a work_dir on setup and this feels like advertising that. 

Also note that it's currently not the same as the arguments to `model.setup`: parameterset is missing.